scheduler/scheduler_medium_test.go

```diff
@@ -54,6 +55,7 @@ func (m *mockMetricManager) CollectMetrics(string, map[string]map[string]string)
 }
 
 func (m *mockMetricManager) PublishMetrics([]core.Metric, map[string]ctypes.ConfigValue, string, string, int) []error {
+	time.Sleep(m.timeToWait)
```

Comment:
Even that is only for testing purpose, I suggest moving this time.Sleep() to CollectMetrics rather than having that in the last step in the workflow which is PublishMetrics. The another reason is that collection normally takes longer than publishing, what seems to be more accurate to reality.